src/commands/commerce/store/quickstart/setup.ts

```diff
@@ -515,17 +538,21 @@ export class StoreQuickstartSetup extends SfdxCommand {
                 this.varargs['communityExperienceBundleName'] as string
             }ToDeploy.zip" ./*`
         );
-        shellJsonSfdx(
-            appendCommonFlags(
-                `sfdx force:mdapi:deploy -u "${this.org.getUsername()}" -g -f "${
-                    this.storeDir
-                }/experience-bundle-package/${
-                    this.varargs['communityExperienceBundleName'] as string
-                }ToDeploy.zip" --wait 60 --verbose --singlepackage`,
-                this.flags,
-                this.logger
-            )
-        );
+
+        let deployCommand = `sfdx force:mdapi:deploy -u "${this.org.getUsername()}" -g -f "${
+            this.storeDir
+        }/experience-bundle-package/${
+            this.varargs['communityExperienceBundleName'] as string
+        }ToDeploy.zip" --wait 60 --verbose --singlepackage`;
+        try {
+            shellJsonSfdx(appendCommonFlags(deployCommand, this.flags, this.logger));
+        } catch (e) {
+            if (JSON.stringify(e.message).indexOf(msgs.getMessage('quickstart.setup.updateSelfRegProfileError')) >= 0) {
+                // if failed with "Error: You can only select profiles that are associated with the experience"
+                // try one more time, it seems to work after second deploy, race issue?
```

Comment:
Do we know the cause of this issue?

Reply:
It happens sporadically if `updateSelfRegProfile` hasn't finished and the profiles aren't yet available in the org. Adding a try/catch and checking for the error seems to help the store successfully create 100% of the time.

Reply:
ok, figured this one out.

When the store update deploy fails in `addContactPointAndDeploy`, `updateSelfRegProfile` was still being called since it was outside of the try/catch. Because the store update deploy failed `<networkMemberGroups>` had not been updated with the guest buyer, which is required before `<selfRegProfile>` is set to the buyer profile.

When `updateSelfRegProfile` attempts the next deploy, it's adding the `<selfRegProfile>` block with the buyer profile but you can't add the <selfRegProfile> block at the same you are assigning the buyer to `<networkMemberGroups>` (which we are doing when the store update fails).

I updated `addContactPointAndDeploy` so that `updateSelfRegProfile` is only called inside the try catch, when we know the previous deploy was successful. I also removed the try/catch from `updateSelfRegProfile`; it's no longer needed since we've got the correct sequence of events now.